### PR TITLE
Fixed order of updates - doesn't always reroute to home on logout

### PR DESCRIPTION
**Pull request by monotkate**
So this is pretty minor. 
(1) I realized that I'd goofed the order of updates on one of my fixes so I fixed that, the most recent update now presents at the top.

(2) I was getting really irritated during testing that it always re-routed to home on logout.  What I added was refreshing the page if there was potentially edits going on during a logout which should wipe any data left unfinished by reloading the controller.  It will reload ideaView and reroute addidea to home.  Any other page it will leave untouched since is far as I'm aware there should be no accessible data that would need to be cleared.  If people don't like this idea or I'm missing something, just let me know.  

@YashdalfTheGray @Mctalian @alanlgirard

### CODE REVIEW

**Comment by Mctalian** [comment]
As far as the update order, I think we were discussing it during your PR
that it'd be better to have the Updates in chronological order because
updates build off each other, much like the KickStarter updates. Making
each update important, not just the most recent.

**Comment by monotkate** [comment]
No, Yash ended up saying that we should care about all the updates, but they should be in reverse chronological order.  I just checked to make sure.

**Comment by Mctalian** [comment]
Hmm memory didn't serve me correct. Either way, David and I found an issue
with deleting updates when there were a few updates, the original author
couldn't delete their own, we might have changed the order to get our
bearings to fix it then forgot to switch it back. So just make sure
everything functions as expected and we'll be good!

**Comment by monotkate** [comment]
Alright, I'll check it probably tomorrow.  Switched comps to work on my other project for now.  I altered things to delete in the right order when they displayed in reverse order, but I think one of the merge conflicts made me change the time tag and accidentally get the order wrong.  I think once the order's fixed, the delete should work correctly

File: src/users/accountView/accountView.js [hunk 1]
```diff
@@ -28,7 +28,13 @@ angular.module('flintAndSteel')
...
                 toastSvc.show(accountName + ' has been logged out!', { duration: 5000 });
 
-                $state.go('home');
+                // Reload if editing data to clear partial content
+                if ($state.includes('idea')) {
+                    $state.reload('idea');
+                }
+                else if ($state.includes('addidea')) {
+                    $state.go('home');
+                }
```
Comment (Mctalian) [inline src/users/accountView/accountView.js]: If we're logging out from the accountView, I think we're already in the "account" state. So I think we actually do want to go "home" from here.

**Comment by Mctalian** [comment]
I changed how updates are deleted on the client-side. It no longer relies on the index of the update, but rather passes in the update object itself. You can then order updates however you want without changing your deleting algorithm.

So don't worry about it anymore, we just need to make sure either my PR goes in first, or mine is shortly after yours or I think the build will break.

**Comment by monotkate** [comment]
Gotcha. Mine should be a quick check. Not a lot of code added. I'm not getting on a computer tonight though and idk about tomorrow.

**Comment by Mctalian** [comment]
Yeah this looks good. Take a look at my comment on accountView when you get a chance, other than that, I approve.

File: src/users/accountView/accountView.js (hunk 1 above)
Comment (monotkate) [inline src/users/accountView/accountView.js]: Good call.  Didn't think that one through just did a copy/paste.  Posting an updated commit for that in a sec

**Comment by monotkate** [comment]
Can I get another approve on this? @YashdalfTheGray @alanlgirard

**Comment by YashdalfTheGray** [comment]
Approved!

**Pull request merged**